### ISSUE TITLE
Half-implemented variant Tifinagh glyphs at inappropriate code points

### ISSUE THREAD

**Issue by dscorbett** [notofonts/tifinagh#8]
### Font
NotoSansTifinaghAPT-Regular.otf
NotoSansTifinaghAdrar-Regular.otf
NotoSansTifinaghAgrawImazighen-Regular.otf
NotoSansTifinaghAhaggar-Regular.otf
NotoSansTifinaghAir-Regular.otf
NotoSansTifinaghAzawagh-Regular.otf
NotoSansTifinaghGhat-Regular.otf
NotoSansTifinaghHawad-Regular.otf
NotoSansTifinaghRhissaIxa-Regular.otf
NotoSansTifinaghSIL-Regular.otf
NotoSansTifinaghTawellemmet-Regular.otf

### Where the font came from, and when
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAPT-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAdrar-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAgrawImazighen-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAhaggar-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAir-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghAzawagh-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghGhat-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghHawad-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghRhissaIxa-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghSIL-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansTifinagh/NotoSansTifinaghTawellemmet-Regular.otf
Date: 2020-04-11

### Font version
Version 2.002

### Issue
All of the Noto Tifinagh fonts except Noto Sans Tifinagh itself attempt to use alternative glyphs for some characters. They only work when the direction is right-to-left.

Furthermore, some of the variant glyphs are so different from the defaults as to be different characters in Unicode’s model of Tifinagh, where unrelated glyphs with the same sound get different code points (e.g. U+2D4A YAZH vs. U+2D4B AHAGGAR YAZH vs. U+2D4C TUAREG YAZH) and a single code point can represent different sounds that share a single glyph (e.g. U+2D40 YAH = Tuareg yab). If a certain glyph isn’t a glyph variant of any already in Unicode’s code chart, it should be proposed to Unicode as a new character.

### Character data
This is just an example. Many characters are affected.

ⴻ
U+2D3B TIFINAGH LETTER YEY

### Screenshots
These screenshots are from Noto Sans Tifinagh APT. The same sort of problem appears in the other fonts for other code points.

LTR:
![ⴻ](https://user-images.githubusercontent.com/1124347/79053612-b8cb3b80-7c0c-11ea-9ce7-87ab590903e3.png)
RTL:
![‮ⴻ‬](https://user-images.githubusercontent.com/1124347/79053613-b8cb3b80-7c0c-11ea-8d6d-64dbacc5bfdf.png)

**Comment by simoncozens** [comment]
I think this is fixed now?

```
simon@Simons-MacBook-Air ~/work/noto/tifinagh .*$ hb-shape master_ttf/NotoSansTifinagh-Regular.ttf 'ⴻ' --direction=ltr
[yey-tifi=0+560]
simon@Simons-MacBook-Air ~/work/noto/tifinagh .*$ hb-shape master_ttf/NotoSansTifinagh-Regular.ttf 'ⴻ' --direction=rtl
[yey-tifi=0+560]
```

**Comment by dscorbett** [comment]
Is it fixed in NotoSansTifinaghAPT-Regular.ttf?

**Comment by simoncozens** [comment]
Ah, no. I didn't quite see the "except Noto Sans Tifinagh itself".

**Comment by simoncozens** [comment]
There is a much bigger problem here. The glyph renaming, to produce the variant glyphs, has never really worked in the fontmake pipeline, only in Glyphs exports. This is because I only actually implemented the "Rename Glyphs" custom parameter in glyphsLib last November. Additionally, the glyph renaming rules are completely mad, and refer to glyphs that we don't even have in the glyphset, such as:

* (In Tewellemmet) `yan_yaf_yat-tifi=yaf_yat-tifi`, `yaw-titi=yawBerber-tifi` (clearly a typo for `yaw-tifi`)
* (In Adrar) `yaghh-tifi=yagAyer-tifi` (*presumably* a typo for `yaghAyer-tifi`)

I've fixed the obvious problems here and now we are building variant glyphs, but I feel out of my depth here and need someone with more familiarity of the different Tifinagh variants to take a look over.

Also the fact that this has been totally broken for years and nobody seems to have noticed is slightly concerning.

**Comment by digitalheir** [comment]
> this has been totally broken for years and nobody seems to have noticed is slightly concerning

Perhaps it is not so strange; many Tamazight / Berber speakers cannot write, those who can mostly make do with what they have. Usually that means: transliterating to Arabic / Latin. 

Capable input methods are only slowly coming along and entering conciousness.

> take a look over

Does it refer specifically to commit f9d7e56052739fa3663a2e980c07a6ca0dd6da78 and all the name changes? Or also the appropriateness of glyph shapes per dialect / variant? 

Some things to note, at least:

- Some names are deceptively similar (yad / yaf fricative / yadj / yaḍ / yaḍ fricative); they are easy to mix up
- Different dialect may use the same shape for different letters
- Glyphs may wildly differ from writer to writer within a dialect, since they have generally not been rigidly formalized with the exception of Moroccan IRCAM letters (Neo-Tifinagh).

**Comment by simoncozens** [comment]
>  Or also the appropriateness of glyph shapes per dialect / variant?

Yes, this; basically, I would like some to review the current fonts, now that they are building with dialectal variants, to ensure that they are doing the right thing and showing the right glyphs for the right letters.

**Comment by abdelhaqueidali** [comment]
It seems there is still some issues with ligatures, i have created a new issue about it. Check it out here, am new to github, so i probably didn't provide enough information in it.
